deploy/staging/application/app.tf

```diff
@@ -16,73 +68,103 @@ data "terraform_remote_state" "platform_stage" {
+data "template_file" "solo_stage_tf_ecs_task_def_template_for_application" {
+  template = "${file("containers_definition.json")}"
   vars = {
-    frontend_container_name = var.frontend_container_name
-    ecs_service_name        = var.ecs_service_name
-    docker_image_url        = data.aws_ecr_repository.solo_stage_tf_ecr_repo.repository_url
-    memory                  = var.memory
-    docker_container_port   = var.docker_container_port
-    region                  = var.region
+    backend_container_name    = var.backend_container_name
+    frontend_container_name   = var.frontend_container_name
+    ecs_task_def_service_name = var.ecs_task_def_service_name
+
+    backend_image_url     = data.aws_ecr_repository.solo_stage_tf_ecr_backend_repo.repository_url
+    frontend_image_url    = data.aws_ecr_repository.solo_stage_tf_ecr_frontend_repo.repository_url
+    backend_digest        = data.aws_ecr_image.backend_digest.image_digest
+    frontend_digest       = data.aws_ecr_image.frontend_digest.image_digest
+    memory                = var.memory
+    docker_container_port = var.docker_container_port
+    region                = var.region
+
+    //  Database
+    POSTGRES_DB       = data.aws_ssm_parameter.POSTGRES_DB.arn
+    POSTGRES_USER     = data.aws_ssm_parameter.POSTGRES_USER.arn
+    POSTGRES_PASSWORD = data.aws_ssm_parameter.POSTGRES_PASSWORD.arn
+    POSTGRES_HOST     = data.aws_ssm_parameter.POSTGRES_HOST.arn
+
+    //  Container
+    MAIN_DOMAIN    = data.aws_ssm_parameter.MAIN_DOMAIN.arn
+    NGINX_SSL_CERT = data.aws_ssm_parameter.NGINX_SSL_CERT.arn
+    NGINX_SSL_KEY  = data.aws_ssm_parameter.NGINX_SSL_KEY.arn
+    API_DOMAIN     = data.aws_ssm_parameter.API_DOMAIN.arn
+    AUTH_DOMAIN    = data.aws_ssm_parameter.AUTH_DOMAIN.arn
+    BACKEND_PROXY  = data.aws_ssm_parameter.BACKEND_PROXY.arn
+    SECRET_KEY     = data.aws_ssm_parameter.SECRET_KEY.arn
   }
 }
 
-//1. Create the Task Definition
-resource "aws_ecs_task_definition" "solo_stage_tf_task" {
-  container_definitions = data.template_file.solo_stage_tf_ecs_task_def_template.rendered
-  family                = var.ecs_service_name
+//1. Task Definition
+resource "aws_ecs_task_definition" "solo_stage_tf_ecs_td_application" {
+  container_definitions = data.template_file.solo_stage_tf_ecs_task_def_template_for_application.rendered
+  family                = var.ecs_task_def_service_name
   network_mode          = "awsvpc"
 
   //  Required for Fargate
   requires_compatibilities = ["FARGATE"]
   cpu                      = var.cpu
   memory                   = var.memory
-  task_role_arn            = data.aws_iam_role.ecs_task_execution_role.arn
-  execution_role_arn       = data.aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn            = data.aws_iam_role.ecs_task_role.arn
+  execution_role_arn       = data.aws_iam_role.ecs_task_exe_role.arn
 
   tags = {
-    Name    = "solo_stage_tf_task"
-    Project = "SOLO"
+    Name    = "solo_stage_tf_td_application"
```

Comment:
I think there's a variable for this defined on line 15, it seems fine just hardcoded like this but then do we need the variable?